src/main/java/duke/DukeException.java

```diff
@@ -0,0 +1,52 @@
+package duke;
+
+/**
+ * Extends from Exception to handle different exceptions.
+ * Different exceptions correspond to different message.
+ */
+public class DukeException extends Exception {
+    private final String eMessage;
```

Comment:
Perhaps could change `eMessage` to `errorMessage` to improve readability.

Reply:
Noted! Thank you for your advance! :)